src/containers/AlertContainer.js

```diff
@@ -23,9 +25,15 @@ class AlertContainer extends Component {
       this.setState({ show: true });
     }
   }
+  componentWillReceiveProps(nextProps) {
+    if (isEmpty(nextProps)) {
+      this.setState({ show: false });
+    }
+  }
 
   handleDismiss = () => {
     this.setState({ show: false });
+    this.props.clearAlert();
```

Comment:
I like this callback solution.  Nice work.

Reply:
@adamjarling Do you think there is a way improve this further more? Anytime we create `AlertContainer` we need to create that `clearAlert()` function. Anyway to avoid that?

Reply:
I'm not entirely sure.  I guess it depends on whether we want to the `<AlertContainer />` to handle the logic for hiding/showing itself, or whether we want to wrap `<AlertContainer />` in a conditional block, in its parent component?

The tricky part I found, was combining the dismissible part, within `AlertContainer`.

What seems to make the most sense to you?

Reply:
Or should the logic to clear out a component's errorObject which feeds the `AlertContainer`, live within the calling component?   I could see this go many ways.

Reply:
I think one of the main purposes of breaking Alerts out of redux-store into a separate React component was to give it the full control of the Alert. So keeping the hiding/showing logic within the `AlertContainer` is best. Keeping clearing the `alertObj` within the calling component and passing it as a `prop` into `AlertContainer` works best for now.

And keeping `<AlertContainer />` within a conditional block, what could we use as a condition in here? Because the `alertObj` resides within the hosting component, which does not get updated when we dismiss the alert from within the `AlertContainer`.

> Or should the logic to clear out a component's errorObject which feeds the `AlertContainer`, live within the calling component? I could see this go many ways.

This is what is done in this PR?

Reply:
Yea, I think the way it's set up now, in this PR seems to work.  Let's run with this for now.   Feel free to merge this one in.